spec/lib/granite/action/performing_spec.rb

```diff
@@ -145,33 +148,49 @@ def execute_perform!(*)
 
           validates :login, format: /\A\w+\z/
 
+          after_commit do
+            callback_result << 'Transaction commited'
+          end
+
           private
 
           def execute_perform!(block:)
             block.call
           end
         end
+
+        allow(action).to receive(:callback_result).and_return(callback_result)
         hide_const('ActiveRecord::Base')
         hide_const('ActiveRecord')
       end
 
       context 'without exceptions' do
-        specify { expect(action.perform(block: -> { 121 })).to eq(121) }
+        let(:block) { -> { 121 } }
+
+        it { is_expected.to eq 121 }
+        specify { expect { subject }.to change { callback_result }.to ['Transaction commited'] }
       end
 
       context 'with Granite::Action::Rollback' do
+        let(:block) { -> { klass.new.validate! } }
         let(:klass) do
           stub_class(:data) do
             include ActiveData::Model
             attribute :name, String
             validates :name, presence: true
           end
         end
-        specify { expect(action.perform(block: -> { klass.new.validate! })).to equal(false) }
+
+        it { is_expected.to eq false }
+        specify { expect { subject }.not_to change { callback_result } }
       end
 
       context 'with exception' do
-        specify { expect { action.perform(block: -> { ActiveRecord::Base }) }.to raise_error(NameError, 'uninitialized constant ActiveRecord') }
+        let(:block) { -> { ActiveRecord::Base } }
+
+        specify do
+          expect { subject }.to raise_error(NameError, 'uninitialized constant ActiveRecord').and not_change { callback_result }
+        end
```

Comment:
This example is aggregated, the ones on L184-185 are separate. Probably use the same approach?